src/main/java/oshi/software/os/mac/local/GlobalMemory.java

```diff
@@ -8,41 +8,68 @@
 package oshi.software.os.mac.local;
 
 import oshi.hardware.Memory;
-import oshi.util.ExecutingCommand;
+import oshi.software.os.mac.local.SystemB.VM_STATISTICS;
+import oshi.software.os.mac.local.SystemB.VM_STATISTICS64;
+
+import com.sun.jna.Platform;
+import com.sun.jna.ptr.IntByReference;
+import com.sun.jna.ptr.LongByReference;
 
 /**
- * Memory obtained by GlobalMemoryStatusEx.
+ * Memory obtained by host_statistics (vm_stat) and sysctl
  * 
- * @author alessandro[at]perucchi[dot]org
+ * @author widdis[at]gmail[dot]com
  */
 public class GlobalMemory implements Memory {
 
 	private long totalMemory = 0;
 
 	public long getAvailable() {
-		long returnCurrentUsageMemory = 0;
+		long availableMemory = 0;
 		long pageSize = 4096;
-		for (String line : ExecutingCommand.runNative("vm_stat")) {
-			if (line.matches("\\D+(page size of \\d+ bytes)\\D+")) {
-				pageSize = new Long(line.replaceAll("\\D", ""));
-			} else if (line.startsWith("Pages free:")
-					|| line.startsWith("Pages speculative:")
-					|| line.startsWith("Pages inactive:")) {
-				String[] memorySplit = line.split(":\\s+");
-				if (memorySplit.length > 1) {
-					returnCurrentUsageMemory += new Long(
-							memorySplit[1].replace(".", ""));
-				}
+
+		int machPort = SystemB.INSTANCE.mach_host_self();
+
+		LongByReference pPageSize = new LongByReference();
+		int ret = SystemB.INSTANCE.host_page_size(machPort, pPageSize);
+		if (ret == 0) {
+			pageSize = pPageSize.getValue();
+		}
+
+		if (Platform.is64Bit()) {
+			VM_STATISTICS64 vmStats = new VM_STATISTICS64();

```

Comment:
Isn't `VM_STATISTICS64` an extension of `VM_STATISTICS`? In that case you could inherit or compose one with the other, then the `availableMemory` math could be the same.